### PR TITLE
feat: assign & seek for `RecordStream` as a fluent API

### PR DESCRIPTION
**Pull request by Kazark**
In a new fluent interface, so as not to make breaking changes, and to avoid
explosion of overloads and accompanying boilerplate. Some functional acrobatics
are needed, but if I'm not mistaken it should be a pleasure to use.

### CODE REVIEW

File: core/src/main/scala/com/banno/kafka/RecordStream.scala [hunk 1]
```diff
@@ -260,7 +260,114 @@ object RecordStream {
...
+    }
+
+    implicit def applyInstance[F[_]]: Apply[Seeker[F, *]] =
+      new Apply[Seeker[F, *]] {
+        override def ap[A, B](
+          ff: Seeker[F, A => B]
+        )(
+          fa: Seeker[F, A]
+        ): Seeker[F, B] =
+          ???
```
Comment (amohrland) [inline core/src/main/scala/com/banno/kafka/RecordStream.scala:310]: I'm guessing this is a to-do?

**Review by amohrland — APPROVED** [review]
I can't claim to understand the acrobatics, but looking forward to trying it out!

File: core/src/main/scala/com/banno/kafka/RecordStream.scala (hunk 1 above)
Comment (Kazark) [reply core/src/main/scala/com/banno/kafka/RecordStream.scala:310]: Ope!

Comment (Kazark) [reply core/src/main/scala/com/banno/kafka/RecordStream.scala]: 21201ae

**Review by zcox — APPROVED** [review]

**Pull request merged**